lib/cisco_node_utils/fabricpath_global.rb

```diff
@@ -209,9 +202,9 @@ def default_linkup_delay_always
     end
 
     def linkup_delay_enable
-      enabled = config_get('fabricpath', 'linkup_delay_enable')
-      return default_linkup_delay_enable if enabled.nil?
-      enabled.first.strip == 'Enabled' ? true : false
+      delay = config_get('fabricpath', 'linkup_delay_enable')
```

Comment:
This is a gotcha with boolean kind and auto-default. The missing match causes a nil and therefore forces the use of default value. If the default value is "true" we get an undesired result.

This is the complete code (as seen below, we want to return false for no match. NOTE: the command used is not a "show run"):
    def linkup_delay_enable
      delay = config_get('fabricpath', 'linkup_delay_enable')
      return false if delay.nil?
      delay
    end


From: Mike Wiebe <notifications@github.com<mailto:notifications@github.com>>
Reply-To: cisco/cisco-network-node-utils <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Friday, November 13, 2015 at 8:41 AM
To: cisco/cisco-network-node-utils <cisco-network-node-utils@noreply.github.com<mailto:cisco-network-node-utils@noreply.github.com>>
Cc: Cisco Employee <dcherian@cisco.com<mailto:dcherian@cisco.com>>
Subject: Re: [cisco-network-node-utils] adding fabricpath topo and fixed feature_set as a class method (#75)


In lib/cisco_node_utils/fabricpath_global.rb<https://github.com/cisco/cisco-network-node-utils/pull/75#discussion_r44803678>:

> @@ -209,9 +202,9 @@ def default_linkup_delay_always
>      end
>
>      def linkup_delay_enable
> -      enabled = config_get('fabricpath', 'linkup_delay_enable')
> -      return default_linkup_delay_enable if enabled.nil?
> -      enabled.first.strip == 'Enabled' ? true : false
> +      delay = config_get('fabricpath', 'linkup_delay_enable')


Why does this property need to useauto_default?

-
Reply to this email directly or view it on GitHub<https://github.com/cisco/cisco-network-node-utils/pull/75/files#r44803678>.